open-company/open-company-web
# Autosave sync

Card: https://trello.com/c/70GuYBKj

Related PRs:
- [Storage](https://github.com/open-company/open-company-storage/pull/228)
- ~~[Change](https://github.com/open-company/open-company-change/pull/24)~~
- ~~[Lib](https://github.com/open-company/open-company-lib/pull/79)~~

To test concurrent edits:
- open 2 sessions of the same user
- with session 1 create a new draft
- with session 2 go to drafts and start editing the same post
- make the edit you make in one session are shown in the other session

To test client race condition on post publish:
- open QP
- type some body and headline quickly
- click Post as soon as you see Saving...
- [x] the post was published?
- [x] no twin post (same headline and body but different uuid) in drafts?

NB: this could have weird outcome in case of users trying to do collaborative editing, for example one user could override the edits of the other. The last edit wins (updated-at is used to compare).
